collect_app/src/main/java/org/odk/collect/android/storage/StorageMigrator.java

```diff
@@ -0,0 +1,201 @@
+package org.odk.collect.android.storage;
+
+import android.content.ContentValues;
+import android.database.Cursor;
+
+import androidx.lifecycle.LiveData;
+import androidx.work.WorkInfo;
+import androidx.work.WorkManager;
+
+import org.apache.commons.io.FileUtils;
+import org.odk.collect.android.dao.FormsDao;
+import org.odk.collect.android.dao.InstancesDao;
+import org.odk.collect.android.dao.ItemsetDao;
+import org.odk.collect.android.database.ItemsetDbAdapter;
+import org.odk.collect.android.forms.Form;
+import org.odk.collect.android.instances.Instance;
+import org.odk.collect.android.itemsets.Itemset;
+import org.odk.collect.android.provider.FormsProvider;
+import org.odk.collect.android.provider.InstanceProvider;
+import org.odk.collect.android.tasks.ServerPollingJob;
+import org.odk.collect.android.upload.AutoSendWorker;
+
+import java.io.File;
+import java.util.List;
+
+import timber.log.Timber;
+
+import static android.provider.BaseColumns._ID;
+import static org.odk.collect.android.database.ItemsetDbAdapter.KEY_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.FORM_FILE_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.FORM_MEDIA_PATH;
+import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.JRCACHE_FILE_PATH;
+import static org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns.INSTANCE_FILE_PATH;
+
+public class StorageMigrator {
```

Comment:
What are you thinking for tests for this? Feels to me like testing the `StorageMigrator` in a all the possible scenarios (each result) and then maybe an Espresso test for the "happy case" (where everything goes right)?

Reply:
Ok I added such a test.
When it comes to Espresso we need to have the UI implemented first.